src/CMakeLists.txt

```diff
@@ -18,7 +18,7 @@ else(MSVC)
   if(NOT WIN32)
     target_compile_options(ada INTERFACE -fPIC)
   endif()
-  target_compile_options(ada PRIVATE -Wall -Wextra -Weffc++)
+  target_compile_options(ada PRIVATE -Wall -Wextra -Weffc++ -Winline)
```

Comment:
I added `-Winline` but GCC12 complains then about the destructor of URL not being inlined. I think that's not a valid warning (the constructor isn't even being called as far as I can see). I would just remove  `-Winline`  for now.

Reply:
In any case, I think you can merge this PR.